timber/src/main/java/timber/log/Timber.java

```diff
@@ -241,19 +241,37 @@ private Timber() {
     private static final Pattern ANONYMOUS_CLASS = Pattern.compile("\\$\\d+$");
     private static final ThreadLocal<String> NEXT_TAG = new ThreadLocal<String>();
 
-    private static String createTag() {
+    /**
+     * Returns an explicitly set tag for the next log message or {@code null}. Calling this method
+     * clears any set tag so it may only be called once.
+     */
+    protected String nextTag() {
       String tag = NEXT_TAG.get();
       if (tag != null) {
         NEXT_TAG.remove();
+      }
+      return tag;
+    }
+
+    /**
+     * Creates a tag for a log message.
+     * <p>
+     * By default this method will check {@link #nextTag()} for an explicit tag. If there is no
+     * explicit tag, the class name of the caller will be used by inspecting the stack trace of the
+     * current thread.
+     */
+    protected String createTag() {
+      String tag = nextTag();
+      if (tag != null) {
         return tag;
       }
 
-      StackTraceElement[] stackTrace = new Throwable().getStackTrace();
-      if (stackTrace.length < 6) {
+      StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+      if (stackTrace.length < 7) {
```

Comment:
I think it'd be slightly better, to write it as
```
stackTrace.length <= 6
```
to make the relation between this value and the '6' in line 274 clearer. Next good step would be extracting the number to a constant variable.